### ISSUE TITLE
Wrong emails pass validation

### ISSUE THREAD

**Issue by niccord** [RotherOSS/otobo#637]
Hi everyone,
if you set "test@gmail .com" as a recipient in your ticket reply, the validation passes but the email send returns an error.

This is due to the Perl `Email::Valid->address( $Param{Address} )` function in `Kernel/System/CheckItem.pm` which considers the email valid.

The Perl functions return `undef` if the email is not valid, a string with the "validated" email otherwise, so as the in example above it returns "test@gmail.com".

As a temporary workaround, I propose to check if the output of the function is different from the input address and throw an error in that case.
Something like: 
``` Perl
my $validatedEmail = Email::Valid->address( $Param{Address} );
if ( !$validatedEmail or $validatedEmail ne $Param{Address} ) {
    $Error = "Invalid syntax";
    $Self->{ErrorType} = 'InvalidSyntax';
}
```

You should also consider changing the behaviour of the `CheckEmail` function, returning the validated email address, in a future version.

Edit by @bschmalhofer : Please note the space before the "." in  "test@gmail .com". I starred at the issue for 5 minutes without seeing the space.

**Comment by svenoe** [comment]
Interesting, this got auto closed... Anyways, thanks, here, too.

**Comment by bschmalhofer** [comment]
After the merge I get some failures when running _CheckItem.t_ under Docker. 
```

ok 27 - CheckEmail() - modperl-uc.1384763750.ffhelkebjhfdihihkbce-michiel.beijen=otrs.com@perl.apache.org
ok 28 - CheckEmail() - test@home.com (Test)
not ok 29 - CheckEmail() - "Test Test" <test@home.com>
not ok 30 - CheckEmail() - "Test Test" <test@home.com> (Test)
not ok 31 - CheckEmail() - Test <test@home(Test).com>
ok 32 - CheckEmail() - <test@home.com
ok 33 - CheckEmail() - test@home.com>
ok 34 - CheckEmail() - test@home.com(Test)
ok 35 - CheckEmail() - test@home.com>(Test)
not ok 36 - CheckEmail() - Test <test@home.com> (Test)
ok 37 - TrimTest

```
If the semantics of CheckEmail() have changed, then the tests should be adapted too.

**Comment by svenoe** [comment]
No...it was a mistake on my side, I guess... I was too fast. I'm pretty sure Email::Valid->address returns only the email address without the name, thus this version just doesn't work. I will change it. Thanks for pointing it out.

**Comment by svenoe** [comment]
Ok, I reverted the pull request for now. I first thought I'd fix it with using
`$Param{Address} !~ /\Q$ValidatedEmail\E/`
and it worked for some cases, but 'test@home(Something).com' still didn't work (it is interpreted as 'test@home.com'). I wasn't aware of this, but seemingly those addresses (also 'test @home.com') are actually considered valid, or at least seem to automatically be convertable, just as Email::Valid->address does it.

In the end, I guess the standard solution here is just not to not accept those addresses, but to convert them when sending the email (it is done for the '(comment)', which is just stripped, I just tested this), so we definitely should change this not in the validation but for the sending itself.

**Comment by stefanhaerter** [comment]
I did some debugging and found several things:

- Part of the problem is `Mail::Address`, which is used e.g. in `Kernel/Modules/AgentTicketCompose.pm` and already removes the space before the @
- The bug with the spaces seems to be already known (https://github.com/Perl-Email-Project/Email-Valid/issues/4) and the suggestion to compare the returned address with the original one seems to be the accepted workaround

**Comment by stefanhaerter** [comment]
Setting milestone to 10.0.26 to increase visibility of this issue.

**Comment by bschmalhofer** [comment]
> In the end, I guess the standard solution here is just not to not accept those addresses, but to convert them when sending the email (it is done for the '(comment)', which is just stripped, I just tested this), so we definitely should change this not in the validation but for the sending itself.

I think that this is already the case in OTOBO 11.0.x . When sending the mail via SMTP the sanitized email addresses, in our example "test@gmail .com" are used for the routing. More precisi The unsanitized addresses are used in the 'To:' header of the MIME message. My impression it that this was already the case in OTOBO 10.0.x. See https://github.com/RotherOSS/otobo/blob/475dc88d10cb9056c9887559aa6b16b16b80b5c5/Kernel/System/Email.pm#L606 .

I tested this quickly with the greenmail SMTP server. There I could reply to "Andy.Admin@asdf .com" and got the Email in the "Andy.Admin@asdf.com" folder.

It would be interesting to see the error that occurred when sending the Email.

**Comment by bschmalhofer** [comment]
See also https://github.com/Perl-Email-Project/Email-Valid/issues/36#issuecomment-3096242875 and https://github.com/RotherOSS/otobo/discussions/4449 .

**Comment by svenoe** [comment]
Closing the issue - I don't know when it was fixed, but I can send to `test@gmail .com` - the address is validated as ok _and_ replaced now.
